### PR TITLE
Migrate to TCA v 1.0.0

### PR DESCRIPTION
**Pull request by thepearl**

### CODE REVIEW

**Comment by pitt500** [comment]
Hi @thepearl, thanks so much for your time migrating the project to 1.0! 🚀

I will take a look at your changes and prepare a backup branch for 0.54 before merging.

File: OnlineStoreTCA/CartList/Cart/CartCell.swift [hunk 1]
```diff
@@ -69,18 +69,3 @@ struct CartCell: View {
         }
     }
 }
-
-struct CartCell_Previews: PreviewProvider {
```
Comment (pitt500) [inline OnlineStoreTCA/CartList/Cart/CartCell.swift]: I'm curious about why all previews were removed from this PR. Is it because of anything related to TCA?

File: OnlineStoreTCA/CartList/CartListDomain.swift [hunk 2]
```diff
@@ -8,15 +8,16 @@
...
     struct State: Equatable {
         var dataLoadingStatus = DataLoadingStatus.notStarted
         var cartItems: IdentifiedArrayOf<CartItemDomain.State> = []
         var totalPrice: Double = 0.0
         var isPayButtonDisable = false
-        var confirmationAlert: AlertState<Action>?
-        var errorAlert: AlertState<Action>?
-        var successAlert: AlertState<Action>?
+        
+        @PresentationState var confirmationAlert: AlertState<Action>?
```
Comment (pitt500) [inline OnlineStoreTCA/CartList/CartListDomain.swift:18]: I'm not aware of the latest TCA changes due to a lot of things in my agenda. What is @PresentationState?

File: OnlineStoreTCA/CartList/CartListView.swift [hunk 3]
```diff
@@ -12,7 +12,7 @@ struct CartListView: View {
     let store: Store<CartListDomain.State, CartListDomain.Action>
     
     var body: some View {
-        WithViewStore(self.store) { viewStore in
+        WithViewStore(self.store, observe: { $0 }) { viewStore in
```
Comment (pitt500) [inline OnlineStoreTCA/CartList/CartListView.swift:15]: What is observe?

File: OnlineStoreTCA/Root/RootDomain.swift [hunk 4]
```diff
@@ -38,7 +38,21 @@ struct RootDomain: ReducerProtocol {
         uuid: { UUID() }
     )
     
-    var body: some ReducerProtocol<State, Action> {
+    static let demo = Self(
```
Comment (pitt500) [inline OnlineStoreTCA/Root/RootDomain.swift]: We do have live and test instances in the project already. I don't think we need these new declarations.

File: OnlineStoreTCA/CartList/CartListView.swift (hunk 3 above)
Comment (thepearl) [reply OnlineStoreTCA/CartList/CartListView.swift:15]: doc:[ init(_:observe:) ](https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/viewstore/init(_:observe:)-3ak1y)
TL;DR the old constructor but they added the observe store as an argument.

File: OnlineStoreTCA/CartList/CartListDomain.swift (hunk 2 above)
Comment (thepearl) [reply OnlineStoreTCA/CartList/CartListDomain.swift:18]: The new way to handle alerts and confirmation dialogue, a great example & explanation [here](https://github.com/pointfreeco/swift-composable-architecture/blob/main/Examples/CaseStudies/SwiftUICaseStudies/01-GettingStarted-AlertsAndConfirmationDialogs.swift) by stephencelis.

File: OnlineStoreTCA/CartList/Cart/CartCell.swift (hunk 1 above)
Comment (thepearl) [reply OnlineStoreTCA/CartList/Cart/CartCell.swift]: I am using xcode beta which has some problems with previews,. i will put them back today in a new commit.

Comment (thepearl) [reply OnlineStoreTCA/CartList/Cart/CartCell.swift]: Previews restored.

Comment (pitt500) [reply OnlineStoreTCA/CartList/Cart/CartCell.swift]: Thanks!

File: OnlineStoreTCA/CartList/CartListDomain.swift (hunk 2 above)
Comment (pitt500) [reply OnlineStoreTCA/CartList/CartListDomain.swift:18]: Thank you, I definitely will take a look!

File: OnlineStoreTCA/CartList/CartListView.swift (hunk 3 above)
Comment (pitt500) [reply OnlineStoreTCA/CartList/CartListView.swift:15]: gotcha!

**Comment by pitt500** [comment]
Hi @thepearl, thank you for addressing the comments. Let me run the code in my local and if everything goes well, I will proceed to merging.

Sounds good?

**Comment by thepearl** [comment]
@pitt500 Sounds awesome !

**Comment by pitt500** [comment]
@thepearl My sincerest apologies for the significant delay in reviewing your PR. It’s been a year, and I truly appreciate your patience. I’ve since updated the codebase to 1.14, so I’ll close this PR. If there’s anything I can do to make up for the time and effort you put into this, please don’t hesitate to let me know.
